### ISSUE TITLE
Use Fraud Pipeline as additional data source for IP address and email address blocks

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#11447]
From the [Account Submission Restrictions PRD](https://docs.google.com/document/d/1FfFcydrcm_TC4sQHXWQfJg2_11RvvYUANDy0hqMlH7U/edit#).

We'll be blocking submission by IP address and email address on https://github.com/mozilla/addons-server/issues/11441 and https://github.com/mozilla/addons-server/issues/11443, respectively. These addresses will be stored on the AMO DB, but the Firefox Accounts Fraud Pipeline contains more data on suspicious account behavior. We should use that data to deal with spammers more effectively.

One possibility would be to query a service when an IP or email check is done, in addition to checking the DB. Another would be to consume the reports from that system and add them automatically to the DB. The first approach is probably better in order to keep the data centralized.

**Comment by diox** [comment]
Filed https://bugzilla.mozilla.org/show_bug.cgi?id=1557742 in case there is private information to share to set this up.

**Comment by diox** [comment]
Setting as qa: not needed for now because it needs some additional setup by ops before we can test it.

**Comment by diox** [comment]
This is ready for testing on dev (stage needs an additional deploy to catch latest ops changes).

QA: let me know when you're ready to test this. It needs someone to insert arbitrary restrictions against some ips / emails into the service to test, and we have no UI for this (since it's all automatic based on the fraud detection pipeline) so I'll have to do it for you.

**Comment by AlexandraMoga** [comment]
Verified on -dev and stage (with @diox 's help) by adding several emails and IPs to the restriction list.
Users will be blocked during submission on the -dev agreement page by the following errors:
- when an IP is blocked: _Multiple add-ons violating our policies have been submitted from your location. The IP address has been blocked._
- when an email is blocked: _The email address you used for your developer account is not allowed for add-on submission._

The restriction also applies when the submission comes from the signing api.

We've also made sure that the restriction is removed once the user reputation comes back to normal.
